lib/coach/middleware_item.rb

```diff
@@ -10,13 +10,15 @@ class MiddlewareItem
     def initialize(middleware, config = {})
       @middleware = middleware
       @config_value = config
+      @block = Proc.new if block_given?
```

Comment:
This is absolute witchcraft - `Proc.new` will use the given block if present, without having to do `&block` in the method signature.